src/Microsoft.Health.Fhir.SqlServer/Features/Schema/Migrations/3.diff.sql

```diff
@@ -166,16 +166,15 @@ BEGIN
     SET NOCOUNT ON
 
     DECLARE @maxSchemaVersion int,
-            @currentSchemaVersion int,
             @minSchemaVersion int
 
-    Select @maxSchemaVersion = min(MaxVersion), @currentSchemaVersion = CurrentVersion, @minSchemaVersion = min(MinVersion)
+    Select @maxSchemaVersion = MIN(MaxVersion), @minSchemaVersion = MAX(MinVersion)
     FROM dbo.InstanceSchema
-    WHERE Timeout > SYSUTCDATETIME() GROUP BY CurrentVersion
+    WHERE Timeout > SYSUTCDATETIME()
 
     SELECT @minSchemaVersion, MAX(Version)
     FROM dbo.SchemaVersion
-    WHERE Status = 'complete' AND Version BETWEEN @currentSchemaVersion AND @maxSchemaVersion
+    WHERE Status = 'complete' AND Version <= @maxSchemaVersion
```

Comment:
Nit: it might be nice to have a comment explaining that we need to ensure that the `max` value we are returning is not greater than the most recent, complete schema version